Public/Src/Cache/ContentStore/Library/Service/LocalContentServerBase.cs

```diff
@@ -450,7 +450,10 @@ protected override async Task<BoolResult> ShutdownCoreAsync(OperationContext con
 
             _portDisposer?.Dispose();
 
-            await _grpcServer.KillAsync();
+            if (_grpcServer != null)
+            {
+                await _grpcServer.KillAsync();

```

Comment:
does null coalescing not work with async?

Reply:
await _grpcServer?.KillAsync() will throw NRE if the field is null because this is equivalent to:

await (_grpcServer != nul ? _grpcServer.KillAsync() : null);

Meaning the code will be transformed into 'await null'.

---
In reply to: [285773484](https://github.com/microsoft/BuildXL/pull/346#discussion_r285773484) [](ancestors = 285773484)